File: frontend/src/types/receipt.ts
export interface Receipt {
  id: number;
  category_id: number | null;
  payer_id: number | null;
  quarter_id: number;
  description: string;
  income: number;
  expense: number;
  discount: number;
  people_count: number;
  receipt_url: string | null;
  is_transferred: boolean;
  transaction_at: string;
  transferred_at: string | null;
}

export interface ReceiptCreate {
  quarter_id: number;
  category_id: number | null;
  payer_id: number | null;
  description: string;
  income: number;
  expense: number;
  discount: number;
  people_count: number;
  receipt_url: string | null;
  is_transferred: boolean;
  transaction_at: string;
  transferred_at: string | null;
}

export interface ReceiptUpdate extends ReceiptCreate {}

File: app/models/receipt.py
from datetime import datetime
from pydantic import BaseModel

class ReceiptRead(BaseModel):
    id: int
    category_id: int | None
    payer_id: int
    description: str
    income: int = 0
    expense: int = 0
    discount: int = 0
    people_count: int = 1
    receipt_url: str | None
    is_transferred: bool = False
    transaction_at: datetime
    transferred_at: datetime | None

class ReceiptCreate(BaseModel):
    category_id: int | None
    payer_id: int
    description: str
    income: int = 0
    expense: int = 0
    discount: int = 0
    people_count: int = 1
    receipt_url: str | None
    is_transferred: bool = False
    transaction_at: datetime
    transferred_at: datetime | None